### ISSUE TITLE
Accessibility Issue with <noscript>

### ISSUE THREAD

**Issue by Frenchomatic** [ampproject/amphtml#14215]
It appears that use of `<noscript>` in the AMP page header breaks WCAG level 2.0 AA for accessibility. Specifically I can see a warning pop up on AMP pages using WebAIM.org . It is the only warning I get if the rest of the page is designed correctly. Removing `<noscript>` and there is a problem in the AMP validator.  How can an AMP page be made accessible? Am I missing an easy trick?

**Comment by Frenchomatic** [comment]
Here we go http://wave.webaim.org/report#/https://www.dentistcodes.com/east-dallas-tx.html - as you can see you get a warning about noscript. The problem is that if you remove it then the page is not AMP valid. In reality this is not likely to be an issue for handicapped users but there is a legal question for healthcare websites and governmental ones which are meant to comply with WCAG level 2.0 AA. Very tenuous I admit but why is AMP using noscript anyway.

**Comment by aghassemi** [comment]
Hi @Frenchomatic AMP needs to support `<noscript>` in the boilerplate CSS so that the page is displayed immediately to users that have javascript disabled (For users who have JS enabled, page is initially hidden until some JS runs first).

Regarding WCAG requirements, I fully understand but this is one of the cases where an automated a11y check can not simply know why `<noscript>` is used and has to do a warning that `<noscript>` should not be considered an alternative for A11Y. 

Our usage of `<noscript>` is fully compliant with actual WCAG requirements even if automated A11Y checkers get it wrong.

**Comment by Frenchomatic** [comment]
Thanks for clearing that up. As I said, accessibility is important for a significant pat of the web industry and particularly so in the healthcare profession. 


On 10 Apr 2018, at 01:01, Ali Ghassemi <notifications@github.com> wrote:

Hi @Frenchomatic <https://github.com/Frenchomatic> AMP needs to support <noscript> in the boilerplate CSS so that the page is displayed immediately to users that have javascript disabled (For users who have JS enabled, page is initially hidden until some JS runs first).

Regarding WCAG requirements, I fully understand but this is one of the cases where an automated a11y check can not simply know why <noscript> is used and has to do a warning that <noscript> should not be considered an alternative for A11Y.

Our usage of <noscript> is fully compliant with actual WCAG requirements even if automated A11Y checkers get it wrong.

**Comment by aghassemi** [comment]
@Frenchomatic Absolutely and we value accessibility a lot in AMP. I have personally worked with Federation of the Blind regarding web a11y in the past and would be happy to be a point of contact if you have any a11y concerns or find a11y bugs in AMP. Always feel free to ping me on issues.
